### PR TITLE
refactor event sinks to support metrics and better logging

### PR DESCRIPTION
**Pull request by ckmcd**

### CODE REVIEW

File: modules/common/src/main/scala/surge/streams/EventSink.scala [hunk 1]
```diff
@@ -28,3 +36,11 @@ trait EventSink[Event] extends EventHandler[Event] {
     FlowConverter.flowFor(handleEvent, partitionBy, sinkExceptionHandler, parallelism)(ExecutionContext.global)
   }
 }
+
+trait EventSinkSupport[Event] {
+  def metrics: Metrics = Metrics.globalMetricRegistry
+  def metricTags: Map[String, String] = Map("class" -> getClass.getCanonicalName)
+  def formatMetaForExceptionHandlerMessage[Meta]: EventPlusStreamMeta[String, Event, Meta] => String = _.streamMeta.toString
```
Comment (jeffboutotte) [inline modules/common/src/main/scala/surge/streams/EventSink.scala:43]: It seems like we could just add these directly to `EventHandler`. Metrics stuff seems like it should be available for any `EventHandler` as we add more observability going forward and an `EventHandler` that does not extend `EventSinkSupport` would get the same implementation as this default anyways via the `DefaultDataSinkExceptionHandler`.

File: modules/common/src/test/scala/surge/internal/health/matchers/RepeatingSignalMatcherSpec.scala [hunk 2]
```diff
@@ -63,8 +63,8 @@ class RepeatingSignalMatcherSpec extends TestKit(ActorSystem("RepeatingSignals")
         Range(1, 100),
         Range(1, 100))
 
-      Source(repeatingData)
-        .mapAsync(parallelism = 10)(data =>
+      Source(repeatingData.asInstanceOf[immutable.Iterable[Range]])
```
Comment (jeffboutotte) [inline modules/common/src/test/scala/surge/internal/health/matchers/RepeatingSignalMatcherSpec.scala:66]: Cleaning up a warning here I'm guessing? Maybe following the pattern in the Akka Streams docs for [`Source.from`](https://doc.akka.io/docs/akka/current/stream/operators/Source/from.html) would be better here? Even though this is just tests, `asInstanceOf` makes me nervous in general

**Review by jeffboutotte — COMMENTED** [review]
Awesome ideas, making things more observable is always better!

File: modules/common/src/main/scala/surge/streams/EventSink.scala (hunk 1 above)
Comment (jeffboutotte) [reply modules/common/src/main/scala/surge/streams/EventSink.scala:43]: Chatted live on this and we'll keep it the way it is to preserve backwards compatibility for now at least. We can look at if it makes sense to revisit later

**Comment by ckmcd** [comment]
No longer needed.

**Pull request closed without merging**